align_anything/evaluation/benchmarks/mt_bench/vllm_eval.py

```diff
@@ -0,0 +1,342 @@
+                    template.user_prompt.format(input=item['instruction'][1]) + \
+                    template.assistant_prompt.format(output="") \
+                    for response_r1, item in zip(responses_r1, data)]
+            else:
+                return [template.system_prompt + template.user_prompt.format(input=item['instruction'][0]) + template.assistant_prompt.format(output="") for item in data]
+    
+    def load_dataset_round2(self, outputs_r1: Dict[str, List[InferenceOutput]]):
+        processed_inputs = {}
+        for task in self.task_names:
+            current_file_path = os.path.abspath(__file__)
+            current_dir = os.path.dirname(current_file_path)
+            dataset = load_dataset(os.path.join(current_dir, self.data_cfgs.task_dir))
+            responses_r1 = [output_r1.response[0] for output_r1 in outputs_r1[task]]
+            prompts, token_ids = self.preprocess(data=dataset, responses_r1=responses_r1)
+            processed_inputs[task] = [InferenceInput(text=prompt, token_ids=token_id) for prompt, token_id in zip(prompts, token_ids['input_ids'])]
+        return processed_inputs
+    
+    def preprocess(self, data, responses_r1=None):
+        prompts = self.build_prompt(data[self.split], responses_r1)
+        token_ids = self.tokenizer(prompts)
+        return prompts, token_ids
+
+class MTBenchGeneratorVLLM(BaseInferencer_vllm):
+    def eval(self, data:Dict[str, List[InferenceInput]], eval_configs) -> Dict[str, List[InferenceOutput]]:
+        task2details = {}
+        for task, input in data.items():
+            task2details[task] = self.generation(input)
+        
+        output_dir = eval_configs.output_dir
+        brief_filename = eval_configs.brief_filename
+        model_id = self.model_cfgs.model_id
+        detailed_filename = f'{model_id}_detailed'
+        brief_filename = f'{model_id}_brief'
+        update_results(output_dir, brief_filename, detailed_filename,task2details)
+        
+        return task2details
+
+def fill_prompt_template(prompt_template, **kwargs):
+    return prompt_template.format(**kwargs)
+
+def get_score(response: str):
+    pattern = r'\[\[(-?\d+(?:\.\d+)?)\]\]'
+    match = re.search(pattern, response)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+class API_Eval(API_Single_Eval):
+    def build_gpt_input(self, judge_prompt: str, user_prompt : str):
+        input = [{'role': 'system', 'content': judge_prompt},
+                {'role': 'user', 'content': user_prompt}]
+    
+        return input
+
+
+def evaluator(raw_output1: List[InferenceOutput], raw_output2: List[InferenceOutput], dataloader: MTBenchDataLoader, task: str, eval_configs= None):
+    current_file_path = os.path.abspath(__file__)
+    current_dir = os.path.dirname(current_file_path)
+    dataset = load_dataset(task, data_files=os.path.join(current_dir, eval_configs.task_dir))[dataloader.split]
```

Comment:
```suggestion
    dataset = load_dataset(current_dir,task)[dataloader.split]
    dataset = load_dataset(current_dir,split='train',data_files='test.jsonl')
```
The first line: load dataset from the current directory, and use the value of `dataloader.split` to get dataset name, if your test prompt file name: `test.jsonl`, it will success, but if your test prompt files name: `test_prompt.jsonl`, it will fail.

The second file: a more robust dataset-load method.

Both of the above commands are correct. The original command leads to ERROR.

Reply:
It is nice of you to provide such detailed feedback on the code! I really appreciate your insights and suggestions.